docs/notebooks/burger.ipynb

```diff
@@ -0,0 +1,755 @@
+    "        \"\"\"Copy data from one storage to another.\n",
+    "\n",
+    "        Parameters\n",
+    "        ----------\n",
+    "        step : int\n",
+    "            The time step.\n",
+    "        from_storage : StorageType\n",
+    "            The storage type from which the data will be copied.\n",
+    "        to_storage : StorageType\n",
+    "            The storage type to which the data will be copied.\n",
+    "        move : bool, optional\n",
+    "            Whether the data will be moved or not. If `True`, the data will be\n",
+    "            removed from the `from_storage`.\n",
+    "        \"\"\"\n",
+    "        if from_storage == StorageType.DISK:\n",
+    "            if step in self.adjoint_dependency[StorageType.DISK]:\n",
+    "                file_name = self.adjoint_dependency[StorageType.DISK][step]\n",
+    "                with open(file_name, \"rb\") as f:\n",
+    "                    self.adjoint_dependency[to_storage][step] = np.load(f)\n",
+    "            if step in self.restart_forward[StorageType.DISK]:\n",
+    "                file_name = self.restart_forward[StorageType.DISK][step]\n",
+    "                with open(file_name, \"rb\") as f:\n",
+    "                    self.forward_work_memory[to_storage][step] = np.load(f)\n",
+    "            if move:\n",
+    "                os.remove(file_name)\n",
+    "        elif from_storage == StorageType.RAM:\n",
+    "            self.forward_work_memory[to_storage][step] = copy.deepcopy(self.restart_forward[from_storage][step])\n",
+    "            if move:\n",
+    "                if step in self.adjoint_dependency[from_storage]:\n",
+    "                    del self.adjoint_dependency[from_storage][step]\n",
+    "                if step in self.restart_forward[from_storage]:\n",
+    "                    del self.restart_forward[from_storage][step]\n",
+    "\n",
+    "    def _initialize_adjoint(self):\n",
+    "        J = self._jacobian(self.forward_final_solution, self.model[\"max_n\"], self.model[\"max_n\"])\n",
+    "        M = self._mass_matrix()\n",
+    "        self.adjoint_work_memory[StorageType.WORK][self.model[\"max_n\"]] = spsolve(J.T, - self.forward_final_solution)\n",
+    "\n",
+    "    def _jacobian(self, u, i, j):\n",
+    "        M = self._mass_matrix()\n",
+    "        if i == j:\n",
+    "            K = self._stiffness_matrix()\n",
+    "            C = self._jacobian_advection(u)\n",
+    "            Jac = M + self.model[\"dt\"] * (- K + C)\n",
+    "        if j == i - 1:\n",
+    "            Jac = - M\n",
+    "        Jac[0, :] = Jac[-1, :] = 0\n",
+    "        Jac[0, 0] = Jac[-1, -1] = 1\n",
+    "        return Jac\n",
+    "    \n",
+    "    def _solve_newton(self, u_prev, residual, tol=1e-8, max_iter=50):\n",
+    "        i = 0\n",
+    "        u_prev[0] = u_prev[-1] = 0\n",
+    "        u = copy.deepcopy(u_prev)\n",
```

Comment:
```suggestion
    "        u = copy.deepcopy(u_prev)\n",
    "        u[0] = u[-1] = 0\n",
```

Reply:
I lose the correct convergence rate with the modification asked above.

Reply:
See previous comment re. the Newton solver residual.